src/quicklist.c

```diff
@@ -374,24 +374,28 @@ REDIS_STATIC void __quicklistCompress(const quicklist *quicklist,
     if (!in_depth)
         quicklistCompressNode(node);
 
+
     /* At this point, forward and reverse are one node beyond depth */
     quicklistCompressNode(forward);
     quicklistCompressNode(reverse);
 }
 
 #define quicklistCompress(_ql, _node)                                          \
     do {                                                                       \
-        if ((_node)->recompress)                                               \
+        if ((_node)->recompress) {                                             \
```

Comment:
maybe in addition to making sure we don't compress the head or tail, we should also have an assertion in the pop and push functions, which assume the head and tail are uncompressed.
i.e. this would catch a case where a node was removed as a whole, and something that was previously not a head, became a head?

Reply:
@oranagra , do you mean add assert(node->encoding != QUICKLIST_NODE_ENCODING_LZF) in push and pop ?

Reply:
yes.. it crashes there anyway, but seeing this assertion will make it easier to figure out the problem.